src/modules/Elsa.Elasticsearch/Features/ElasticsearchFeature.cs

```diff
@@ -1,59 +1,67 @@
 using Elastic.Clients.Elasticsearch;
 using Elsa.Elasticsearch.Extensions;
 using Elsa.Elasticsearch.HostedServices;
-using Elsa.Elasticsearch.Models;
+using Elsa.Elasticsearch.Modules.Management;
+using Elsa.Elasticsearch.Modules.Runtime;
 using Elsa.Elasticsearch.Options;
 using Elsa.Elasticsearch.Services;
 using Elsa.Features.Abstractions;
 using Elsa.Features.Services;
+using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
 
 namespace Elsa.Elasticsearch.Features;
 
+/// <summary>
+/// Configures Elasticsearch.
+/// </summary>
 public class ElasticsearchFeature : FeatureBase
 {
+    /// <inheritdoc />
     public ElasticsearchFeature(IModule module) : base(module)
     {
     }
-    
-    internal ElasticsearchOptions Options { get; set; } = new();
-    internal IDictionary<Type, string> IndexConfig { get; set; } = IElasticConfiguration.GetDefaultIndexConfig();
-    internal IndexRolloverStrategy? IndexRolloverStrategy { get; set; }
 
+    /// <summary>
+    /// A delegate that configures Elasticsearch.
+    /// </summary>
+    public Action<ElasticsearchOptions> Options { get; set; } = _ => { };
+
+    /// <summary>
+    /// True to enable index name rollovers, false otherwise.
+    /// </summary>
+    public bool EnableRollover { get; set; }
+
+    /// <inheritdoc />
     public override void ConfigureHostedServices()
     {
-        Module.ConfigureHostedService<ConfigureMappingHostedService>(-1);
+        Module.ConfigureHostedService<ConfigureElasticsearchClientHostedService>(-2);
```

Comment:
I think this line can be also under if EnableRollover condition as it is creating an alias which is redundant if a rollover is not being used.

Also, ConfigureElasticsearchClientHostedService can be renamed to ConfigureAliasesHostedService as it is the only responsibility of the service.

Reply:
Makes sense, I'll rename it 👍🏻